janusgraph-core/src/main/java/org/janusgraph/graphdb/management/JanusGraphManager.java

```diff
@@ -0,0 +1,199 @@
+    }
+
+    public Set<String> getGraphNames() {
+        return graphs.keySet();
+    }
+
+    public Graph getGraph(String gName) {
+        return graphs.get(gName);
+    }
+
+    public void putGraph(String gName, Graph g) {
+        graphs.put(gName, g);
+    }
+
+    /**
+     * @Deprecated
+     */
+    public ConcurrentHashMap getTraversalSources() {
+        return traversalSources;
+    }
+
+    public Set<String> getTraversalSourceNames() {
+        return traversalSources.keySet();
+    }
+
+    public TraversalSource getTraversalSource(String tsName) {
+        return null; // JanusGraph do not support storing references to traversalSources
+    }
+
+    public void putTraversalSource(String tsName, TraversalSource ts) {
+        String errMsg = "JanusGraph does not support TraversalSource reference tracking.";
+        throw new JanusGraphManagerDoesNotManageTraversalSourcesException(errMsg);
+    }
+
+    public TraversalSource removeTraversalSource(String tsName) {
+        String errMsg = "JanusGraph does not support TraversalSource reference tracking.";
+        throw new JanusGraphManagerDoesNotManageTraversalSourcesException(errMsg);
+    }
+
+    public Bindings getAsBindings() {
+        final Bindings bindings = new SimpleBindings();
+        graphs.forEach(bindings::put);
+        return bindings;
+    }
+
+    public void rollbackAll() {
+        graphs.forEach((key, value) -> {
+            final Graph graph = value;
+            if (graph.tx().isOpen()) {
+                graph.tx().rollback();
+            }
+        });
+    }
+
+    public void rollback(final Set<String> graphSourceNamesToCloseTxOn) {
+        commitOrRollback(graphSourceNamesToCloseTxOn, false);
+    }
+
+    public void commitAll() {
+        graphs.forEach((key, value) -> {
+            final Graph graph = value;
+            if (graph.tx().isOpen())
+                graph.tx().commit();
+        });
+    }
+
+    public void commit(final Set<String> graphSourceNamesToCloseTxOn) {
+        commitOrRollback(graphSourceNamesToCloseTxOn, true);
+    }
+
+    public void commitOrRollback(Set<String> graphSourceNamesToCloseTxOn, Boolean commit) {
+        graphSourceNamesToCloseTxOn.forEach(e -> {
+            final Graph graph = getGraph(e);
+            if (null != graph) {
+                closeTx(graph, commit);
+            }
+        });
+    }
+
+    public void closeTx(Graph graph, Boolean commit) {
+        if (graph.tx().isOpen()) {
+            if (commit) {
+                graph.tx().commit();
+            } else {
+                graph.tx().rollback();
+            }
+        }
+    }
+
+    public Graph openGraph(String gName, Function<String, Graph> thunk) {
+        Graph graph = graphs.get(gName);
+        if (graph != null) {
+            return graph;
+        } else {
+            synchronized (instantiateGraphLock) {
+                graph = graphs.get(gName);
+                if (graph == null) {
+                    try {
```

Comment:
Remove try-catch block. Any exceptions will already be unchecked RuntimeExceptions which I don't think you need to wrap.